src/librustc_typeck/astconv.rs

```diff
@@ -88,9 +92,16 @@ struct ConvertedBinding<'tcx> {
     span: Span,
 }
 
-struct ParamRange {
-    required: usize,
-    accepted: usize
+#[derive(PartialEq)]
+enum GenericArgPosition {
+    Datatype,
+    Function,
+    Method,
+}
+
+struct GenericArgMismatchErrorCode {
+    lifetimes: (&'static str, &'static str),
+    types: (&'static str, &'static str),
```

Comment:
Can you open an issue and write a FIXME here, about uniformizing these? Error codes are sad business and in this case, there shouldn't be more than one error.

Reply:
Opened https://github.com/rust-lang/rust/issues/53525.